parsec/core/gui/keys_widget.py

```diff
@@ -0,0 +1,98 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPLv3 2019 Scille SAS
+
+import shutil
+import os
+from pathlib import Path
+
+from PyQt5.QtCore import pyqtSignal
+from PyQt5.QtWidgets import QWidget, QFileDialog
+
+from parsec.core.local_device import (
+    list_available_devices,
+    AvailableDevice,
+    load_device_file,
+    get_devices_dir,
+)
+
+from parsec.core.gui.ui.keys_widget import Ui_KeysWidget
+from parsec.core.gui.ui.key_widget import Ui_KeyWidget
+from parsec.core.gui.custom_dialogs import show_error, ask_question
+from parsec.core.gui.lang import translate
+
+
+class KeyWidget(QWidget, Ui_KeyWidget):
+    export_clicked = pyqtSignal(AvailableDevice)
+
+    def __init__(self, device, parent=None):
+        super().__init__(parent=parent)
+        self.device = device
+        self.setupUi(self)
+        self.label_org.setText(device.organization_id)
+        self.label_device.setText(device.device_label)
+        self.label_user.setText(device.human_handle.label)
+        self.export_button.clicked.connect(self._on_export)
+
+    def _on_export(self):
+        self.export_clicked.emit(self.device)
+
+
+class KeysWidget(QWidget, Ui_KeysWidget):
+    def __init__(self, config, parent):
+        super().__init__(parent=parent)
+        self.config = config
+        self.setupUi(self)
+        self.reload_devices()
+        self.button_import_key.clicked.connect(self._on_import_key)
+
+    def reload_devices(self):
+        layout = self.scroll_content.layout()
+        for _ in range(layout.count()):
+            item = self.scroll_content.layout().takeAt(0)
+            layout.removeItem(item)
+            w = item.widget()
+            if w is not None:
+                w.setParent(None)
+        devices = list_available_devices(self.config.config_dir)
+        for device in devices:
+            w = KeyWidget(device, parent=self)
+            w.export_clicked.connect(self._on_export_key)
+            self.scroll_content.layout().insertWidget(self.scroll_content.layout().count() - 1, w)
+
+    def _on_export_key(self, device):
+        file_dialog = QFileDialog()
+        output_directory = file_dialog.getExistingDirectory()
+        if not output_directory:
+            return
+        _, key_name = os.path.split(device.key_file_path)
+        try:
+            shutil.copyfile(device.key_file_path, os.path.join(output_directory, key_name))
+        except IOError as err:
+            show_error(self, translate("EXPORT_KEY_ERROR"), err)
+
+    def _on_import_key(self):
+        file_dialog = QFileDialog()
+        file_dialog.setFileMode(QFileDialog.ExistingFile)
+        key_file, _ = file_dialog.getOpenFileName(
+            parent=self,
+            caption="Import a key",
+            filter="Keys files (*.keys)",
+            initialFilter="Keys files (*.keys)",
```

Comment:
Those should be translated.

I can't find a reference to `initialFilter` in the documentation (https://doc.qt.io/qt-5/qfiledialog.html#getOpenFileName). I can see what it's supposed to be doing but where does it come from?

Weirdly enough, the flag `QDialogDialog.HideNameFilterDetails` seems to be set by default, which hide the extensions and just shows `Key files` instead of `Key files (*.keys)`. That may be disturbing for a user who renamed their key file.

I think the better option would be to add two filters: one for `*.keys`, one for `*` and set the `initialFilter` (wherever it comes from) to `*.keys`. This way the user still has the option to select any file if their own key file has a different extension.

Reply:
Good idea, 

It's strange that `initalFilter` is not in the documentation because it is clearly defined in the pytqt sip file like that: 
```python
getOpenFileName(parent: QWidget = None, caption: str = '', directory: str = '', filter: str = '', initialFilter: str = '', options: Union[QFileDialog.Options, QFileDialog.Option] = 0)
```